js/background.js

```diff
@@ -6,28 +6,49 @@ chrome.runtime.onInstalled.addListener(function(details){
 });
 
 chrome.runtime.onMessage.addListener(function(request, sender, sendResponse) {
-    if (request.method == "logPageView"){
-		var optOut = localStorage['tracking-opt-out'],
-			deferTracking = false;
-
-		if(optOut && optOut === 'true'){
-			deferTracking = true;
-		}
-
-		if(!deferTracking) {
-			// Standard Google Universal Analytics code
-			(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
-			(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
-			m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
-			})(window,document,'script','https://www.google-analytics.com/analytics.js','ga'); // Note: https protocol here
-
-			ga('create', 'UA-34217520-4', 'auto');
-			ga('set', 'checkProtocolTask', function(){}); // Removes failing protocol check. @see: http://stackoverflow.com/a/22152353/1958200
-			ga('require', 'displayfeatures');
-			ga('send', 'pageview', '/');
-		}
-    }
-    sendResponse({});
+	switch (request.method) {
+		case "logPageView":
+			var optOut = localStorage['tracking-opt-out'],
+				deferTracking = false;
+
+			if(optOut && optOut === 'true'){
+				deferTracking = true;
+			}
+
+			if(!deferTracking) {
+				// Standard Google Universal Analytics code
+				(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+				(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+				m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+				})(window,document,'script','https://www.google-analytics.com/analytics.js','ga'); // Note: https protocol here
+
+				ga('create', 'UA-34217520-4', 'auto');
+				ga('set', 'checkProtocolTask', function(){}); // Removes failing protocol check. @see: http://stackoverflow.com/a/22152353/1958200
+				ga('require', 'displayfeatures');
+				ga('send', 'pageview', '/');
+			}
+			break;
+
+		case "newTab":
+			chrome.tabs.create({
```

Comment:
Ah yeah I just looked at the documentation again: `You can use most chrome.tabs methods and events without declaring any permissions in the extension's manifest file. However, if you require access to the url, title, or favIconUrl properties of tabs.Tab, you must declare the "tabs" permission in the manifest, as shown below:`